### ISSUE TITLE
performance analysis

### ISSUE THREAD

**Issue by eroux** [buda-base/editserv#11]
logging in on library-dev.bdrc.io then going to my user profile triggers a request to

http://editserv.bdrc.io/resource-nc/user/me

which took no less than 10s the first time... second time was 200ms which is better, but I'm thinking maybe we could have some perf analysis... what's the underlying sparql query?

**Comment by MarcAgate** [comment]
Actually, in the first case, you prabably had a auth0 profile but no BudaUser profile. Creating a Budauser profile requires:

1) To create the Buda user rdf model from Auth0 User 
2) To publish this model (i.e do a pull from remote then commit to local and push to remote
3) To publish the public part of the model to the public dataset
4) To publish the whole buda user rdf model to authrw
5) to return the json serialization of the model

All git operations are a serious bottleneck 

In the second case, only one request to authrw is required.

**Comment by eroux** [comment]
yes, git is a bottleneck but it doesn't really need to be... you can push and pull every 10mn max so that the bottleneck is not as strong... see the code around here for an example:

https://github.com/buda-base/buda-iiif-presentation/blob/newbiblio/src/main/java/io/bdrc/iiif/presentation/resservices/BVMService.java#L49

**Comment by MarcAgate** [comment]
Yes, that's one way to reduce BUDA User creation time. 
We need to pull before committing anyway, in order to avoid all sorts of potential issues, but for push operation, yes, a simple thread would actually suffice: we start it and continue as we don't need the push to be completed for executing the remaining tasks of the overall process. 

Actually, as soon as the new BUDA User rdf models are built, we could even have a pull/commit/push overall "git Thread" running on the side while updating fuseki and returning the user rdf profile on the other side. I am going to try that. 

Let's keep in mind however that performance might not be really critical for user creation (around 100 cases/month over the last 3 years according to Chris).

**Comment by eroux** [comment]
I don't think we need to pull every time no, there's not going to be any kind of meddling with the git repo under normal circumstances so we can make optimistic assumptions... in case we have to meddle, we can be extra careful or even shut down editserv temporarily

**Comment by MarcAgate** [comment]
For our Information, this is the performance I have measured for Buda user creation process (described above) on the server:

**Test1 ----------------------------------------------**
Pulling users repo took 669 ms
Creating buda users models took 196 ms
Git add file took 47 ms
Git commit took 165 ms
Git push took 1458 ms
Updating fuseki dataset after git took 73 ms

`TOTAL: 2608 ms`

**Test2 ----------------------------------------------**
Pulling users repo took 447 ms
Creating buda users models took 27 ms
Git add file took 10 ms
Git commit took 5 ms
Git push took 1512 ms
Updating fuseki dataset after git took 74 ms

`TOTAL: 2075 ms`

**Test3 ----------------------------------------------**
Pulling users repo took 864 ms
Creating buda users models took 27 ms
Git add file took 5 ms
Git commit took 5 ms
Git push took 1353 ms
Updating fuseki dataset after git took 68 ms

`TOTAL : 2322 ms`
